java/core/src/main/java/co/worklytics/psoxy/impl/RESTApiSanitizerImpl.java

```diff
@@ -92,10 +92,10 @@ Map<Endpoint, Pattern> getCompiledAllowedEndpoints() {
         return compiledAllowedEndpoints;
     }
 
-    SchemaRuleUtils.JsonSchemaFilter getRootDefinitions() {
+    JsonSchemaFilterUtils.JsonSchemaFilter getRootDefinitions() {
         if (rootDefinitions == null) {
             synchronized ($writeLock) {
-                rootDefinitions = SchemaRuleUtils.JsonSchemaFilter.builder().definitions(rules.getDefinitions()).build();
+                rootDefinitions = JsonSchemaFilterUtils.JsonSchemaFilter.builder().definitions(rules.getDefinitions()).build();
```

Comment:
for doing the double locking singleton properly you need to check again in the lock

```suggestion
  if (rootDefinitions == null) {
          rootDefinitions = JsonSchemaFilterUtils.JsonSchemaFilter.builder().definitions(rules.getDefinitions()).build();
  }
```